File: speedyGo/src/main/java/speedy/go/speedygo/message/MessageRequest.java
package speedy.go.speedygo.message;

import lombok.*;
import speedy.go.speedygo.models.MessageType;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class MessageRequest {

    private String content;
    private String senderId;
    private String receiverId;
    private MessageType type;
    private String chatId;
    private Double lat;
    private Double lng;
    private String address;
}

File: speedyGo-ui/src/app/chat/models/message-request.ts
export interface MessageRequest {
  chatId?: string;
  content?: string;
  receiverId?: string;
  senderId?: string;
  type?: 'TEXT' | 'IMAGE' | 'VIDEO' | 'AUDIO' | 'LOCATION';
  lat?: number;
  lng?: number;
  address?: string;
}

